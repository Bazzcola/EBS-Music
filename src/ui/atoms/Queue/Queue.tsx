import React from 'react';
import 'ui/atoms/Queue/Queue.scss';
type Image = {
    img:string;
}
export const Queue = (props:Image) => {
    return (
        <div className="queue_box">
            <div className="queue_image" style={{
          backgroundImage:
            `url(
            ${props.img}
            )`,
          backgroundPosition: 'center',
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat'
        }}><span>Queue</span></div>
        </div>
    )
}