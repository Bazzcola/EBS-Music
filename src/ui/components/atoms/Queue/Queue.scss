.queue_box{
    width:100px;
    position:relative;
    border-radius: 12px;
    padding:8px 12px 8px 8px;
    cursor: pointer;
    .queue_image{
        width:28px;
        height:28px;
        box-sizing: border-box;
        border-radius: 3px;
    }
    span{
        position:absolute;
        top: 15px;
        right: 30px;
        font-size: 13px;
        font-family: Open Sans, Arial, sans-serif;
    }
}
.queue_box:hover{
    background:#d1d1d6;
}