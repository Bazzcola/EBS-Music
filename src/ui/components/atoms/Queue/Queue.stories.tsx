import React from 'react';
import { Queue } from 'ui/components/atoms/Queue/Queue';
export default {
  title: 'Atoms/Queue',
  component: Queue
};

export const Button7 = () => {
  const img =
    'https://akket.com/wp-content/uploads/2019/11/World-of-Warcraft-Shadowlands-Blizzard.jpg';
  return <Queue img={img} />;
};
