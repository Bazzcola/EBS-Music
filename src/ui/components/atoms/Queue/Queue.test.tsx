import React from 'react';
import { render } from '@testing-library/react';
import { Queue } from './Queue';
const queue_img =
  'https://akket.com/wp-content/uploads/2019/11/World-of-Warcraft-Shadowlands-Blizzard.jpg';
it('renders queue correctly', () => {
  const { asFragment } = render(<Queue img={queue_img} />);
  expect(asFragment()).toMatchSnapshot();
});
