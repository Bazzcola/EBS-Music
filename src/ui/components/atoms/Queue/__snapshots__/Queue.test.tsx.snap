// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders queue correctly 1`] = `
<DocumentFragment>
  <div
    class="queue_box"
  >
    <div
      class="queue_image"
      style="background-position: center; background-size: cover; background-repeat: no-repeat;"
    >
      <span>
        Queue
      </span>
    </div>
  </div>
</DocumentFragment>
`;
